packages/api/server/models/shantytownModel/getExportHistory.js

```diff
@@ -0,0 +1,81 @@
+const sequelize = require('#db/sequelize');
+const { fromGeoLevelToTableName } = require('#server/utils/geo');
+const { restrict } = require('#server/utils/permission');
+const SQL = require('./_common/SQL');
+
+
+module.exports = async (user, location, lastDate) => {
+    const where = [];
+    const replacements = {
+    };
+
+    const restrictedLocation = restrict(location).for(user).askingTo('list', 'shantytown');
+    if (restrictedLocation === null) {
+        return [];
+    }
+
+    if (restrictedLocation.type !== 'nation') {
+        where.push(`${fromGeoLevelToTableName(restrictedLocation.type)}.code = :shantytownLocationCode`);
+        replacements.shantytownLocationCode = restrictedLocation[restrictedLocation.type].code;
+    }
+    const shantytown_history = await sequelize.query(
+        `
+            SELECT
+                shantytown_history.*
```

Comment:
Après réflexion, j'ai l'impression que la requête peut-être faite de façon à ne retenir QUE la ligne qui nous intéresse, de la façon suivante :
- `select rank() OVER(partition by shantytown_id ORDER BY updated_at DESC) FROM [...] WHERE updated_at <= :lastDate AND rank = 1`